CI failure in TracingTest#testContextPropagationInFaultTolerance

@pavolloffay @kenfinnigan we have CI failing from time to time due to the test added [here](https://github.com/quarkusio/quarkus/pull/2038).

The failure is the following:
```
2019-04-23 11:38:25,393 INFO  [io.quarkus] (main) Quarkus stopped in 0.033s
[ERROR] Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 6.981 s <<< FAILURE! - in io.quarkus.smallrye.opentracing.deployment.TracingTest
[ERROR] testContextPropagationInFaultTolerance  Time elapsed: 0.971 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <5> but was: <4>
	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
	at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:195)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:152)
	at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:147)
	at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:510)
	at io.quarkus.smallrye.opentracing.deployment.TracingTest.testContextPropagationInFaultTolerance(TracingTest.java:105)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```

It happens once in a while. I saw it twice, once on a Linux build and once on a Windows build.

Could you take a look at that issue and, hopefully, fix it before tomorrow's release?

--- a/extensions/smallrye-opentracing/deployment/pom.xml
+++ b/extensions/smallrye-opentracing/deployment/pom.xml
@@ -90,6 +90,11 @@
             <type>test-jar</type>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.awaitility</groupId>
+            <artifactId>awaitility</artifactId>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
     <build>

--- a/extensions/smallrye-opentracing/deployment/src/test/java/io/quarkus/smallrye/opentracing/deployment/TracingTest.java
+++ b/extensions/smallrye-opentracing/deployment/src/test/java/io/quarkus/smallrye/opentracing/deployment/TracingTest.java
@@ -1,7 +1,9 @@
 package io.quarkus.smallrye.opentracing.deployment;
 
 import java.util.List;
+import java.util.concurrent.TimeUnit;
 
+import org.awaitility.Awaitility;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.asset.EmptyAsset;
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
@@ -36,7 +38,6 @@ public class TracingTest {
     @AfterEach
     public void after() {
         mockTracer.reset();
-        System.out.println(GlobalTracer.get());
     }
 
     @AfterAll
@@ -101,7 +102,10 @@ public class TracingTest {
             Response response = RestAssured.when().get("/faultTolerance");
             response.then().statusCode(200);
             Assertions.assertEquals("fallback", response.body().asString());
+            Awaitility.await().atMost(5, TimeUnit.SECONDS)
+                    .until(() -> mockTracer.finishedSpans().size() == 5);
             List<MockSpan> spans = mockTracer.finishedSpans();
+
             Assertions.assertEquals(5, spans.size());
             for (MockSpan mockSpan : spans) {
                 Assertions.assertEquals(spans.get(0).context().traceId(), mockSpan.context().traceId());

